guardian/s3-chunking-lambda
# Bump node and dependencies to node 14

## What does this change?
Upgrade to Node 14 (that's the latest supported by Lambda according to [this page](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html))

While we're here we're also bumping some dependencies.

___

_But wait, what does this lambda do?_

We log metrics from our podcast CDNs (Fastly and Acast) so that we can mimic 'pageviews' in Ophan for our podcasts. These logs arrive in separate buckets. This lambda then copies them into a single bucket, 'chunking' large logs out into several files. Then another lambda, the [podcasts analytics lambda](https://github.com/guardian/podcasts-analytics-lambda), collects and transforms these metrics to send to Ophan. The 'chunking' of this lambda helps stop the analytics lambda from running out of time.

This lambda is therefore crucial to the Ophan data pipeline for podcast analytics.

## How to test
Adjust the build in TeamCity to use the CI script instead of installing its own version of node. If the build succeeds, update the cloudformation template on PROD, and deploy this branch. This lambda is triggered whenever an object is created in the specified fastly and acast log buckets, monitor their activity and see if the lambda works as intended (i.e. running successfully on Node 14).

**UPDATE**: the above has been carried out, and the lambda is working as intended with Node 14.

NB. Once this branch is merged in, I can delete the node build step in TC completely (currently it is just disabled). We will not need to update the cloudformation template as we have already updated it to test the changes.

## How can we measure success?
If it works, success!

## Have we considered potential risks?
If it breaks we'll have to revert the cloudformation and redeploy the previous known good build while we rethink.

## Images
N/A

## Accessibility
N/A
